src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -262,3 +283,54 @@ mp::VMImageInfo mp::LXDVMImageVault::info_for(const mp::Query& query)
 
     throw std::runtime_error(fmt::format("Unable to find an image matching \"{}\"", query.release));
 }
+
+void mp::LXDVMImageVault::poll_download_operation(const QJsonObject& json_reply, const ProgressMonitor& monitor,
+                                                  const TaskCompleteAction& task_complete)
+{
+    if (json_reply["metadata"].toObject()["class"] == QStringLiteral("task") &&
+        json_reply["status_code"].toInt(-1) == 100)
+    {
+        QUrl task_url(QString("%1/operations/%2")
+                          .arg(base_url.toString())
+                          .arg(json_reply["metadata"].toObject()["id"].toString()));
+
+        // Instead of polling, need to use websockets to get events
+        while (true)
+        {
+            try
+            {
+                auto task_reply = mp::lxd_request(manager, "GET", task_url);
+
+                if (task_reply["error_code"].toInt(-1) != 0)
+                {
+                    break;
+                }
+
+                auto status_code = task_reply["metadata"].toObject()["status_code"].toInt(-1);
+                if (status_code == 200)
+                {
+                    task_complete(task_reply["metadata"].toObject());
+                    break;
+                }
+                else
+                {
+                    auto download_progress = parse_percent_as_int(
+                        task_reply["metadata"].toObject()["metadata"].toObject()["download_progress"].toString());
+
+                    if (!monitor(LaunchProgress::IMAGE, download_progress))
+                    {
+                        mp::lxd_request(manager, "DELETE", task_url);
+                        throw mp::AbortedDownloadException{"Download aborted"};
+                    }
+
+                    std::this_thread::sleep_for(1s);
+                }
```

Comment:
No need for error handling?

Reply:
I'm not sure I understand.  The error handling is above the `else` here.

Reply:
Oh right, forgot it's not only in `status_code`. Shouldn't we log the details of the error, then?

Reply:
Yeah, you're right, we should log the error there.

Reply:
Done